### ISSUE TITLE
Implement ES as an optimizer

### ISSUE THREAD

**Issue by anandtrex** [IGITUGraz/L2L#56]
Since there are papers [1][2] that claim it's the bees knees.

[1] Salimans, Tim, et al. "Evolution Strategies as a Scalable Alternative to Reinforcement Learning." arXiv preprint arXiv:1703.03864 (2017).
[2] Morse, Gregory, and Kenneth O. Stanley. "Simple Evolutionary Optimization Can Rival Stochastic Gradient Descent in Neural Networks." Proceedings of the 2016 on Genetic and Evolutionary Computation Conference. ACM, 2016.
